lib/internal/http2/core.js

```diff
@@ -1077,6 +1077,8 @@ function onHandleFinish() {
   if (this[kID] === undefined) {
     this.once('ready', onHandleFinish.bind(this));
   } else {
+    if (this[kState].headersSent === false)
+      this.rstWithNoError();
```

Comment:
hmm... I don't know if we want to do this. If you look at my latest update, `stream.end()` is called automatically when the request is a `HEAD` request because we know there won't be any payload data sent. This way the user will get a write after end error if they try to write payload data for a `HEAD` request. I'm also catching it in `respond()` so that the `options.endStream` flag is set ensuring that no body can be sent at all.